File: src/main/java/com/galigeigei/acloudmap/entity/model/DataBO.java
package com.galigeigei.acloudmap.entity.model;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

/**
 * 大盘云图个股涨跌数据
 * @author 周士钰
 * @date 2024/9/30 下午9:06
 */
@NoArgsConstructor
@Data
public class DataBO implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * code
     */
    private String code;
    /**
     * name
     */
    private String name;
    /**
     * total
     */
    private long total;
    /**
     * value
     */
    private List<Long> value;
    /**
     * children
     */
    private List<ChildrenDTO> children;

    /**
     * ChildrenDTO
     */
    @NoArgsConstructor
    @Data
    public static class ChildrenDTO {
        /**
         * code
         */
        private String code;
        /**
         * name
         */
        private String name;
        /**
         * total
         */
        private long total;
        /**
         * value
         */
        private List<Long> value;
        /**
         * children
         */
        private List<ChildrenDTO3> children;

        /**
         * ChildrenDTO
         */
        @NoArgsConstructor
        @Data
        public static class ChildrenDTO3 {
            /**
             * id
             */
            private int id;
            /**
             * code
             */
            private String code;
            /**
             * name
             */
            private String name;
            /**
             * price
             */
            private double price;
            /**
             * today
             */
            private String today;
            /**
             * total
             */
            private long total;
            /**
             * value
             */
            private List<Long> value;
            /**
             * arrValue
             */
            private String arrValue;
            /**
             * increase
             */
            private double increase;
            /**
             * intoDate
             */
            private String intoDate;
            /**
             * turnover
             */
            private double turnover;
            /**
             * creatTime
             */
            private String creatTime;
        }
    }

}

File: rustcode/src/models/data_bo.rs
use serde::{Deserialize, Serialize};

/// 大盘云图个股涨跌数据
#[derive(Debug, Serialize, Deserialize)]
pub struct DataBO {
    /// 股票代码
    pub code: String,
    /// 股票名称
    pub name: String,
    /// 总市值
    pub total: f64,
    /// 数据数组
    pub value: Vec<f64>,
    /// 子数据
    pub children: Vec<ChildrenDTO>,
}

/// 子数据
#[derive(Debug, Serialize, Deserialize)]
pub struct ChildrenDTO {
    /// 股票代码
    pub code: String,
    /// 股票名称
    pub name: String,
    /// 总市值
    pub total: f64,
    /// 涨跌幅%
    pub increase: f64,
    /// 换手率/涨跌额
    pub turnover: f64,
    /// 数据数组
    pub value: Vec<f64>,
    /// 子数据
    pub children: Vec<ChildrenDTO3>,
}

/// 子数据3
#[derive(Debug, Serialize, Deserialize)]
pub struct ChildrenDTO3 {
    /// id
    pub id: i32,
    /// 股票代码
    pub code: Option<String>,
    /// 股票名称
    pub name: Option<String>,
    /// 价格
    pub price: Option<f64>,
    /// 今日
    pub today: Option<String>,
    /// 总市值
    pub total: Option<i64>,
    /// 值
    pub value: Option<String>,
    /// 数组值
    pub arr_value: Option<String>,
    /// 涨跌幅%
    pub increase: Option<f64>,
    /// 上市日期
    pub into_date: Option<String>,
    /// 换手率
    pub turnover: Option<f64>,
    /// 创建时间
    pub creat_time: Option<String>,
}

/// 二级板块涨跌数据
#[derive(Debug, Serialize, Deserialize)]
pub struct SectionBO {
    /// 股票代码
    pub code: String,
    /// 股票名称
    pub name: String,
    /// 总市值
    pub total: f64,
    /// 数据数组
    pub value: Vec<f64>,
    /// 子数据
    pub children: Vec<ChildrenDTO>,
}